### ISSUE TITLE
The default trebuchet launcher will reset the home screen after a device reboot when more than 8 apps are pinned to the dock

### ISSUE THREAD

**Issue by Sidefix** [Archfx/duo-de#41]
### Acknowledgements

- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] This is not a feature request.
- [X] This issue is not a duplicate of an existing issue.
- [X] The issue is solely related to this GSI.
- [X] This issue occurs with the latest available build.
- [X] I am not running a custom vendor on my device.
- [X] This issue is not kernel related.
- [X] This issue is not related to Magisk/KernelSU/APatch.

### Info

- Device: Surface Duo 1
- GSI version: v2024.07.25
- Vendor version:

### Expected Behavior

As far as I am aware, there is no limit to the number of items that can be pinned to the multitask bar / dock

### Current Behavior

Currently a maximum of 8 items can be docked. Docking a ninth (regardless of whether it is inside a folder or not) will cause the ENTIRE home screen to reset after a device reboot.

The issue does not occur with 8 items or less. 9+ are required to reproduce the issue.

Screenshot of my home screen, left side only, main page, 8 items:
![20240812_143624 dock 8 items](https://github.com/user-attachments/assets/3fa850d9-a593-4359-94d9-236a6b14da8f)

Screenshot of the same side home screen after rebooting after adding a 9th item
![20240812_145117 screen post reboot 9](https://github.com/user-attachments/assets/cd461e0f-1699-4e17-8453-ec21b1442698)

### Possible Solution

Pin less than 9 items in the dock

### Steps to Reproduce

1. Pin 8 apps to the dock
2. Reboot
3. Notice all good
4. Pin one more app
5. Reboot
6. Notice home screen completely resets

**Comment by Sidefix** [comment]
Quick note, the screenshot of the pre-issue I provided is with 8 items; rebooting with 8 items in the dock will work as intended.

You will need 9+ to reproduce the issue.

**Comment by Sidefix** [comment]
Quick update: I've managed to reproduce the issue in other conditions as well, with the home screen populated with widgets and less than 8 items in the dock... so all of a sudden I'm very confused.

To be clear, I can reliably reproduce the issue as described in the initial issue, however it looks like it can be reproduced in other conditions as well.

**Comment by Sidefix** [comment]
New update... it's become quite frustrating to the point of inusability.

I've had the issue occur many more times with very few reliable repeatable steps. I'd dare say it does it whenever it wants now on a reboot.

I determined to reset the app and clear its data. Then, with the trebuchet data freshly cleared, I DID NOT modify the home screen at all and rebooted to see what it would do.

This also caused it to 'reset' in some weird way where most of the apps simply disappeared similar to the screenshot I had pasted earlier.

In short, the built-in trebuchet launcher seems to be quite bugged to the point of inusability.

**Comment by Archfx** [comment]
This issue is fixed in the latest release.
